### PR TITLE
Add action for document building

### PR DESCRIPTION
**Pull request by wangnuannuan**
1. there is 1 job created in this action
  - job to build HTML 

Before create this action , we need to create a repo secret, because we need to publish the generated files to another repo https://github.com/foss-for-synopsys-dwc-arc-processors/foss-for-synopsys-dwc-arc-processors.github.io.
1. Generate a new **Personal access tokens**
2.  Go to https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain, **Settings->Secrets->New repository secret**, name the secret as **ACTION_TOKEN**, and then copy **Personal access tokens** you create before.

https://github.com/wangnuannuan/toolchain/actions

Signed-off-by: Jingru Wang <jingru@synopsys.com>

### CODE REVIEW

**Comment by wangnuannuan** [comment]
Because we need use `rst2pdf` module, so the version of Sphinx should not newer than `4.*`, there is an action https://github.com/ammaraskar/sphinx-action can be used, but it is based on on container shpinx-doc, an in this container, the version of Sphinx is `4.1.2.`. https://github.com/sphinx-doc/docker/blob/master/base/Dockerfile#L15 So I think it's better to use ubuntu container, and then install Sphinx modules with pip

File: .github/workflows/doc-build.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    - name: cache-pip
+      uses: actions/cache@v1
+      with:
+        path: ~/.cache/pip
+        key: pip-${{ runner.os }}-doc
+
+    - name: install-pip
+      run: |
+        sudo pip3 install -U setuptools wheel pip
+        pip3 install breathe
```
Comment (temap) [inline .github/workflows/doc-build.yml]: @wangnuannuan, could you please create requirements.txt in doc directory with required package versions and call 'pip3 install -r requirements.txt' instead?

File: .github/workflows/doc-build.yml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    - name: cache-pip
+      uses: actions/cache@v1
+      with:
+        path: ~/.cache/pip
+        key: pip-${{ runner.os }}-doc
+
+    - name: install-pip
+      run: |
+        pip3 install -U setuptools wheel pip
+        pip3 install breathe
```
Comment (temap) [inline .github/workflows/doc-build.yml]: the same here

File: .github/workflows/doc-build.yml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+      run: |
+        if [[ "$GITHUB_REF" =~ "refs/tags" ]]; then
+          DOC_TAG="release"
+        else
+          DOC_TAG="development"
+        fi
+        DOC_TAG=${DOC_TAG} SPHINXOPTS="-q -j auto" make -C doc pdf
+
+    - name: upload-build
+      uses: actions/upload-artifact@master
```
Comment (temap) [inline .github/workflows/doc-build.yml]: Please, use actions/upload-artifact@v2 instead of master branch

File: .github/workflows/doc-build.yml [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+        if [[ $(git status --porcelain toolchain | wc -l) != 0 ]]; then
+          git add -A toolchain
+          git commit -s -m "toolchain: Generate pages for commit ${GITHUB_SHA}" --author="svc-arcoss_auto <svc-arcoss_auto@synopsys.com>"
+          git push ${REPO_LINK} master
+        fi
+
+  doc-build-pdf:
+    name: "Documentation Build (PDF)"
+    runs-on: ubuntu-latest
+    container: texlive/texlive:latest
```
Comment (temap) [inline .github/workflows/doc-build.yml]: Do you really need a container here?

File: .github/workflows/doc-build.yml [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    steps:
+    - name: checkout
+      uses: actions/checkout@v2
+
+    - name: install-pkgs
+      run: |
+        sudo apt-get install -y doxygen
+
+    - name: cache-pip
+      uses: actions/cache@v1
```
Comment (temap) [inline .github/workflows/doc-build.yml]: use actions/cache@v2

File: .github/workflows/doc-build.yml [hunk 6]
```diff
@@ -0,0 +1,104 @@
+name: Documentation Build and Deploy
+
+on:
+  push:
+    branchs:
```
Comment (temap) [inline .github/workflows/doc-build.yml]: looks like a typo, should be branches

**Review by Namemrrungphet — APPROVED** [review]

**Review by Namemrrungphet — APPROVED** [review]

**Comment by kolerov** [comment]
It's already done in these accepted PRs:

1. https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/pull/508
2. https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/pull/516

**Pull request closed without merging**